### ISSUE TITLE
Cannot allocate memory in static TLS block

### ISSUE THREAD

**Issue by mbergkvist** [messense/cargo-zigbuild#27]
Hi! Thanks for a great project, it has helped us a lot when building and deploying to AWS Lambda.

I get a similar error as discussed in #4,
`target/release/deps/libthiserror_impl-c21f6147db4ecc92.so: cannot allocate memory in static TLS block`

when I'm building with
`cargo +nightly zigbuild -p lambda --release --target aarch64-unknown-linux-gnu.2.26`

in a Docker container from
```Dockerfile
ARG RUST_VERSION=1.60
FROM rust:$RUST_VERSION as builder

RUN curl "https://ziglang.org/download/0.9.1/zig-linux-aarch64-0.9.1.tar.xz" | tar -J -x -C /usr/local/bin && \
    cargo install cargo-lambda --version 0.5.3 && \
    cargo install cargo-zigbuild --version 0.8.1

FROM rust:$RUST_VERSION
COPY --from=builder /usr/local/cargo/bin/cargo-lambda \
    /usr/local/cargo/bin/cargo-zigbuild \
    /usr/local/cargo/bin/
COPY --from=builder /usr/local/bin/zig-linux-aarch64-0.9.1 \
    /usr/local/bin/zig-linux-aarch64-0.9.1
RUN rustup default nightly-aarch64-unknown-linux-gnu && \
    ln -s /usr/local/bin/zig-linux-aarch64-0.9.1/zig /usr/local/bin/zig
```

Building without explicit glibc-version works fine, both on nightly and stable channel.

I'm not sure what the culprit is. Moving the failing code to another package does not reproduce the error, so I guess it's not the code per se. Any ideas what's the cause? Searching the internet indicates it's related to jemalloc?

I'll see if I can recreate the error with some sample code.

**Comment by messense** [comment]
What's your host machine architecture? #4 only happens when the host triple is the same as the target triple.

**Comment by mbergkvist** [comment]
The host where I'm running the Docker container is `aarch64-apple-darwin` when building locally, and the image when building in CodePipeline is `aws/codebuild/amazonlinux2-aarch64-standard:2.0` so I assume that is `aarch64-unknown-linux-gnu` .

I had to switch to the nightly channel to get the specified glibc-version to be picked up by zig, as mentioned in #4, and that worked fine until I got the `cannot allocate memory in static TLS block` error.

I'm doing `cargo clean` between each build when I'm trying to find out what wrong here, just to make sure to not inherit anything from previous builds.

I only get the error when I specify the glibc version.

`cargo zigbuild -p lambda --release --target aarch64-unknown-linux-gnu` and `cargo +nightly zigbuild -p lambda --release --target aarch64-unknown-linux-gnu` works fine (does not end in any error), only `cargo +nightly zigbuild -p lambda --release --target aarch64-unknown-linux-gnu.2.26` is failing.

**Comment by messense** [comment]
It would be helpful if you can create a minimal reproducer, thanks!

**Comment by mbergkvist** [comment]
I've now shared a stripped down project with you in https://github.com/mbergkvist/allocate-memory-in-static-tls-block.

**Comment by messense** [comment]
Thanks for the report and the minimal reproducer, should be fixed in #31 now. It was a oversight from #19 .

**Comment by mbergkvist** [comment]
Thanks for the swift help. Much appreciated.

**Comment by mbergkvist** [comment]
> should be fixed in #31 now

I've verified that release 0.8.3 also fixed the problem we had in our project.
